sources/engine/Stride.Rendering/Rendering/ComputeEffect/GGXPrefiltering/Hammersley.sdsl

```diff
@@ -12,9 +12,9 @@ namespace Stride.Rendering.Images
         {
             var u = 0.0;
             var p = 0.5;
-            for (int kk=k; kk; p*=0.5, kk>>=1)
+            for (int kk=k; kk > 0; p *= 0.5, kk /= 2)
             {
-                if (kk & 1) // kk mod 2 == 1
+                if (kk % 2 == 1)
                     u += p;
             }
```

Comment:
Can you go over this change, the logic is equivalent of course, but is there a need to move from binary to math operator ?

Reply:
@Eideren Thanks for pointing out. Looks like binary operators works well on GLSL so there is no need to change that since this is faster than division. Long story short there is `HLSLToGlslConvertor` class generated wrong Hammerslay glsl code that did not compile at all.
```glsl
{
    float u = 0.0;
    float p = 0.5;

    for (int kk = k; kk; p *= 0.5, kk >>= 1)
    {
        if (bool(bool(bool(kk & 1))))
            u += p;
    }
    float v = (float(k) + 0.5) / float(samplesCount);
    return vec2(u, v);
}
```
The main issue here is `kk` condition which is not understandable for OpenGL shader langauage

Reply:
So yeah, I'm leaving now only the explicit if more than zero condition